### PR TITLE
[Tools] Namespace check for shared zookeeper

### PR DESCRIPTION
**Pull request by snalli**
Namespaces only exist on zk.shared cluster

### CODE REVIEW

File: ambry-tools/src/main/java/com/github/ambry/clustermap/HelixVcrPopulateTool.java [hunk 1]
```diff
@@ -60,11 +60,18 @@ public static void main(String[] args) throws IOException {
...
-    if (destZkAndCluster.length != 3) {
-      errorAndExit("dest argument must have form 'zkString/namespace/clusterName'");
+    String destZkString;
+    String destClusterName;
+    if (destZkAndCluster[0].contains("shared")) {
+      if (destZkAndCluster.length != 3) {
+        errorAndExit("dest argument must have form 'zkString/namespace/clusterName'");
+      }
+      destZkString = destZkAndCluster[0] + HelixVcrUtil.SEPARATOR + destZkAndCluster[1];
+      destClusterName = destZkAndCluster[2];
```
Comment (justinlin-linkedin) [inline ambry-tools/src/main/java/com/github/ambry/clustermap/HelixVcrPopulateTool.java:70]: I don't think we need namespace in the src and dest zookeeper address. We would just take whatever address provided in the argument. If ambry's zookeeper znode exists under a chroot, we should make sure it's provided in the argument.

Comment (snalli) [reply ambry-tools/src/main/java/com/github/ambry/clustermap/HelixVcrPopulateTool.java:70]: ack

Comment (snalli) [reply ambry-tools/src/main/java/com/github/ambry/clustermap/HelixVcrPopulateTool.java:70]: chroot == namespace.

On old zk clusters, the znode is like zk_string/cluster_name
But on new (shared) zk clusters, the znode is like zk_string/ambry/clustername

clustername is always the last arg.
but the prefix to that varies.

Comment (snalli) [reply ambry-tools/src/main/java/com/github/ambry/clustermap/HelixVcrPopulateTool.java:70]: this came about when working with shared ZK clusters

Comment (snalli) [reply ambry-tools/src/main/java/com/github/ambry/clustermap/HelixVcrPopulateTool.java]: done, changed the code to excl namesapce

File: ambry-tools/src/main/java/com/github/ambry/clustermap/HelixVcrPopulateTool.java [hunk 2]
```diff
@@ -60,11 +60,10 @@ public static void main(String[] args) throws IOException {
...
     String[] destZkAndCluster = options.valueOf(destOpt).split(HelixVcrUtil.SEPARATOR);
-    if (destZkAndCluster.length != 3) {
-      errorAndExit("dest argument must have form 'zkString/namespace/clusterName'");
-    }
-    String destZkString = destZkAndCluster[0] + HelixVcrUtil.SEPARATOR + destZkAndCluster[1];
-    String destClusterName = destZkAndCluster[2];
+    // string could be zk/namespace/clusterName or zk/clusterName
+    int lastSeparator = destZkAndCluster[0].lastIndexOf(SEPARATOR);
+    String destZkString = destZkAndCluster[0].substring(0, lastSeparator);
+    String destClusterName = destZkAndCluster[0].substring(lastSeparator+1);
```
Comment (justinlin-linkedin) [inline ambry-tools/src/main/java/com/github/ambry/clustermap/HelixVcrPopulateTool.java:66]: This is better, but one bug here, it should be
```
String destZkAndCluster = options.valueOf(destOpt);
int lastSep = destZkAndCluster.lastIndexOf(SEPARATOR);
String destZkString = destZkAndCluster.substring(0, lastSep);
String destClusterName = destZkAndCluster.substring(lastSep + 1);
```

File: ambry-tools/src/main/java/com/github/ambry/clustermap/HelixVcrPopulateTool.java [hunk 3]
```diff
@@ -88,11 +87,9 @@ public static void main(String[] args) throws IOException {
...
       if (options.has(srcOpt)) {
         String[] srcZkAndCluster = options.valueOf(srcOpt).split(SEPARATOR);
-        if (srcZkAndCluster.length != 3) {
-          errorAndExit("src argument must have form 'zkString/namespace/clusterName'");
-        }
-        String srcZkString = srcZkAndCluster[0] + HelixVcrUtil.SEPARATOR + srcZkAndCluster[1];
-        String srcClusterName = srcZkAndCluster[2];
+        // string could be zk/namespace/clusterName or zk/clusterName
+        String srcZkString = srcZkAndCluster[0].substring(0, lastSeparator);
+        String srcClusterName = srcZkAndCluster[0].substring(lastSeparator+1);;
```
Comment (justinlin-linkedin) [inline ambry-tools/src/main/java/com/github/ambry/clustermap/HelixVcrPopulateTool.java:92]: same as above.

File: ambry-tools/src/main/java/com/github/ambry/clustermap/HelixVcrPopulateTool.java (hunk 2 above)
Comment (snalli) [reply ambry-tools/src/main/java/com/github/ambry/clustermap/HelixVcrPopulateTool.java]: good catch !

**Review by justinlin-linkedin — APPROVED** [review]

**Pull request merged**